.github/workflows/python-package.yml

```diff
@@ -11,38 +10,24 @@ jobs:
   build-linux-lint:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-python@v2
-      with:
-        python-version: '3.8'
-    - name: Run all tests
-      run: |
-        cat /etc/docker/daemon.json
-        echo '{"insecure-registries" : [ "localhost:5000" ]}' | sudo tee /etc/docker/daemon.json
-        sudo service docker restart
-        sleep 2
-        docker info
-        pip install -e .
-        pip install -r ./lint-requirements.txt
-        flake8
-        isort --check ./
-        black --check ./
-
-  build-linux-test-with-binaries-other-tests:
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-python@v2
-      with:
-        python-version: '3.8'
-    - name: Run all tests
-      run: |
-        bash tests/ci-setup.sh
-        python -m pytest -v --durations=10 --ignore=tests/python_on_whales/components
-
+      - uses: actions/checkout@master
+      - uses: actions/setup-python@master
```

Comment:
You mentionned it was better to use master, but it seems an anti-pattern to unpin versions of software we're using in the CI, as a new version can break our CI without any action from our side. Can you expand a bit on the motivation?

Reply:
I'm actually not set on this change - happy to change to newer versions that don't cause the warnings seen if you scroll down on a run, e.g. [here](https://github.com/gabrieldemarmiesse/python-on-whales/actions/runs/7411813041).

The goal is to avoid the CI breaking due to an external change, as you stated. There are two options with separate reasons it could break:
- Don't pin (use latest/master) and a new version could have a breaking change
- Do pin, and GH actions could drop support for something it depends on

We're seeing warnings because of being on old versions and GH actions no longer supports the version of node the action was written to work with - this could in theory cause the action to break. On the other hand, we use hardly any customisation options with the actions, which are highly unlikely to break in future versions. So it seemed to me perhaps the balance is more in favour of just using latest, but in reality it probably doesn't matter and I was mainly just seeking to get rid of the warnings!

Reply:
In a perfect world, we would pin the version of every dependency/software we use in the CI, and a bot would open pull requests to bump the versions when a new one is available, ensuring we are always up to date and don't use outdated stuff that github might remove. 
This would also ensure that we have perfect reproducibility of our builds, which is always nice. But yeah, we depend on github actions which is a moving target in terms of pre-installed software.

Even if it's not fool-proof, I'd prefer that we stick to best practices and pin the versions. I agree that we might get warnings if we don't change the version for a while, but at least, if on the action side of things, they push a bug in their master branch, our CI won't break, which is a good thing.

long story short, I'd prefer that we just upgrade the versions and do not use master.